doc/tutorials/objdetect/traincascade.markdown

```diff
@@ -184,146 +103,115 @@ img/img2.jpg  2  100 200 50 50   50 30 25 25
+After the opencv\_traincascade application has finished its work, the trained cascade will be saved in `cascade.xml` file in the `-data` folder. Other files in this folder are created for the case of interrupted training, so you may delete them after completion of training.
 
-    -   -weightTrimRate \<weight_trim_rate\>
+Training is finished and you can test your cascade classifier!
 
-        Specifies whether trimming should be used and its weight. A decent choice is 0.95.
+Visualising Cascade Classifiers
+-------------------------------
 
-    -   -maxDepth \<max_depth_of_weak_tree\>
+From time to time it can be usefull to visualise the trained cascade, to see which features it selected and how complex its stages are. For this OpenCV supplies a opencv\_visualisation application. This application has the following commands:
 
-        Maximal depth of a weak tree. A decent choice is 1, that is case of stumps.
+ - `--image` <b>(required)</b> : path to a reference image for your object model. This should be an annotation with dimensions [`-w`,`-h`] as passed to both opencv\_createsamples and opencv\_traincascade application.
+ - `--model` <b>(required)</b> : path to the trained model, which should be in the folder supplied to the `-data` parameter of the opencv\_traincascade application.
+ - `--data` <i>(optional)</i> : if a data folder is supplied, which has to be manually created beforehand, stage output and a video of the features will be stored.
 
-    -   -maxWeakCount \<max_weak_tree_count\>
+An example command can be seen below
 
-        Maximal count of weak trees for every cascade stage. The boosted classifier (stage) will
-        have so many weak trees (\<=maxWeakCount), as needed to achieve the
-        given -maxFalseAlarmRate.
+@code{.text}
+opencv\_visualisation --image=/data/object.png --model=/data/model.xml --data=/data/result/
+@endcode
 
--#  Haar-like feature parameters:
+Some limitations of the current visualisation tool
+ - Only handles cascade classifier models, trained with the opencv\_traincascade tool, containing __stumps__ as decision trees [default settings].
+ - The image provided needs to be a sample window with the original model dimensions, passed to the `--image` parameter.
 
-    -   -mode \<BASIC (default) | CORE | ALL\>
+Example of the HAAR/LBP face model ran on a given window of Angelina Jolie, which had the same preprocessing as cascade classifier files -->24x24 pixel image, grayscale conversion and histogram equalisation:
 
-        Selects the type of Haar features set used in training. BASIC use only upright features,
-        while ALL uses the full set of upright and 45 degree rotated feature set. See @cite Lienhart02
-        for more details.
+_A video is made with for each stage each feature visualised:_
 
--#  Local Binary Patterns parameters:
+![9757_05_14](https://cloud.githubusercontent.com/assets/4621239/14203748/1bef5efc-f800-11e5-84b2-939d47e46161.png)
```

Comment:
Images should be added to the repository, for example to the `opencv/doc/tutorials/objdetect/images/` folder.